bindings/python/tests/test_core.py

```diff
@@ -27,6 +28,16 @@ def test_blocking():
     assert meta.mode.is_file()
     assert [str(entry) for entry in op.list("/")] == ["test"]
     assert [str(entry) for entry in op.scan("/")] == ["test"]
+
+    reader = op.reader("test")
```

Comment:
By the way, is our reader a valid `file-like` object in Python? I expect it to be compatible with all Python functions that accept a `file-like` object.

Reply:
By definition, [`file-like object`](https://docs.python.org/3/glossary.html#term-file-like-object) or [`file object`](https://docs.python.org/3/glossary.html#term-file-object) is an object exposing a file-oriented API (with methods such as `read()` or `write()`) to an underlying resource.

So at least it lacks `write` (and maybe `tell`), we can add them to simply raise `NotImplementedError`.

Reply:
SGTM.

Reply:
> Do we need to implement support for `with op.reader("test")` so that the underlying reader can be automatically dropped when out of scope?

```rust
    pub fn __enter__(slf: Py<Self>) -> Py<Self> {
        slf
    }

    pub fn __exit__(&mut self, _exc_type: PyObject, _exc_value: PyObject, _traceback: PyObject) {}
```

Looks like the best we can do is this, it can't drop anything in `__exit__` since there's no api for it (`drop(self.0)` cannot move out of `self` which is behind a mutable reference).

```python
reader = op.reader('test')
with reader:
    ...

reader.read()  # still works
```

Reply:
Unless we change `BlockingReader` to `struct BlockingReader(Option<od::BlockingReader>)`...

Reply:
> Unless we change `BlockingReader` to `struct BlockingReader(Option<od::BlockingReader>)`...

LGTM, the only way